src/assets/wise5/authoringTool/node/node-authoring/node-authoring.component.ts

```diff
@@ -292,37 +292,56 @@ export class NodeAuthoringComponent implements OnInit {
     this.hideComponentAuthoring();
   }
 
-  protected deleteButtonClicked(): void {
+  protected deleteComponents(): void {
     this.scrollToTopOfPage();
     this.hideComponentAuthoring();
-
-    let confirmMessage = '';
-    const selectedComponentNumbersAndTypes = this.getSelectedComponentNumbersAndTypes();
-    if (selectedComponentNumbersAndTypes.length == 1) {
-      confirmMessage = $localize`Are you sure you want to delete this component?\n`;
-    } else if (selectedComponentNumbersAndTypes.length > 1) {
-      confirmMessage = $localize`Are you sure you want to delete these components?\n`;
-    }
-    for (let c = 0; c < selectedComponentNumbersAndTypes.length; c++) {
-      confirmMessage += '\n' + selectedComponentNumbersAndTypes[c];
-    }
-    if (confirm(confirmMessage)) {
-      const selectedComponents = this.getSelectedComponentIds();
-      const data = {
-        componentsDeleted: this.getComponentObjectsForEventData(selectedComponents)
-      };
-      for (const componentId of selectedComponents) {
-        this.projectService.deleteComponent(this.nodeId, componentId);
-      }
-      this.saveEvent('componentDeleted', 'Authoring', data);
-      this.checkIfNeedToShowNodeSaveOrNodeSubmitButtons();
-      this.projectService.saveProject();
+    if (this.confirmDeleteComponent(this.getSelectedComponentNumbersAndTypes())) {
+      const selectedComponentIds = this.getSelectedComponentIds();
+      const deletedComponents = selectedComponentIds.map((componentId) => {
+        return this.projectService.deleteComponent(this.nodeId, componentId);
+      });
+      const componentIdAndTypes = deletedComponents.map((component) => {
+        return { componentId: component.id, type: component.type };
+      });
```

Comment:
Chain pipe operators for brevity? You also don't need a ```return``` if it's just one line and the line returns a value.
```suggestion
      const componentIdAndTypes = this.getSelectedComponentIds()
        .map((componentId) => this.projectService.deleteComponent(this.nodeId, componentId))
        .map((component) => ({ componentId: component.id, type: component.type }));
```